katharsis-core/src/test/java/io/katharsis/jackson/RelationshipContainerSerializerTest.java

```diff
@@ -430,7 +431,40 @@ public void onMultipleNestedIncludes() throws Exception {
         // confirm default variables are referenced in second level nested includes
         assertThatJson(result).node("included[6].relationships.projectEager.data").isEqualTo(null);
         assertThatJson(result).node("included[6].relationships.tasks.data").isArray();
+    }
+
+    @Test
+    public void onRelationshipLookUpWithNestedIncludes() throws Exception {
+        QueryParamsBuilder queryParamsBuilder = new QueryParamsBuilder(new DefaultQueryParamsParser());
+        Map<String, Set<String>> queryParamsMap = new HashMap<>();
+        queryParamsMap.put("include[projects]", new HashSet<>(Collections.singletonList("task.projectsInit")));
+        QueryParams queryParams = queryParamsBuilder.buildQueryParams(queryParamsMap);
+
+        Project includedProject = new Project().setId(2L);
```

Comment:
You know, you could just add a constructor to the `Project` class. :) Then all could benefit. If you do, remember to keep a default one for jackson